### PR TITLE
Add Graph Mode Dynamic Quant tutorial

### PR DESCRIPTION
**Pull request by supriyar**
Summary:
Tutorial to demonstrate graph mode dynamic quant on BERT model.
Currently not directly runnable as it requires to download glue dataset and fine-tuned model

### CODE REVIEW

**Comment by jlin27** [comment]
@supriyar - Here's a direct preview link to your tutorial: https://deploy-preview-1065--pytorch-tutorials-preview.netlify.app/prototype/graph_mode_dynamic_bert_tutorial.html

Initial pass looks good. I'll take a deeper dive later today

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+
+This tutorial introduces the steps to do post training Dynamic Quantization with Graph Mode Quantization. Dynamic quantization converts a float model to a quantized model with static int8 data types for the weights and dynamic quantization for the activations. The activations are quantized dynamically (per batch) to int8 while the weights are statically quantized to int8. Graph Mode Quantization flow operates on the model graph and requires minimal user intervention to quantize the model. To be able to use graph mode, the float model needs to be either traced or scripted first.
+
+Advantages of graph mode quantization are:
+
+- In graph mode, we can inspect the code that is executed in forward function (e.g. aten function calls) and quantization is achieved by module and graph manipulations.
+- Simple quantization flow, minimal manual steps.
+- Unlocks the possibility of doing higher level optimizations like automatic precision selection.
+
+For additional details on Graph Mode Quantization please refer to the Graph Mode Static Quantization Tutorial.
```
Comment (jlin27) [inline prototype_source/graph_mode_dynamic_bert_tutorial.rst:18]: Suggestion: Add hyperlink for Graph Mode Static Quantization Tutorial

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    qconfig_dict = {'': per_channel_dynamic_qconfig}
+
+qconfig is a named tuple of the observers for activation and weight. For dynamic quantization we use a dummy activation observer to mimic the dynamic quantization process that happens in the operator during runtime. For the weight tensors we recommend using per-channel quantization which helps improve the final accuracy.
+``qconfig_dict`` is a dictionary with names of sub modules as key and qconfig for that module as value, empty key means the qconfig will be applied to whole model unless it’s overwritten by more specific configurations, the qconfig for each module is either found in the dictionary or fallback to the qconfig of parent module.
+
+Right now qconfig_dict is only way to configure how the model is quantized, and it is done in the granularity of module, that is, we only support one type of qconfig for each module, and the qconfig for sub module will override the qconfig for parent module. for example, if we have
```
Comment (jlin27) [inline prototype_source/graph_mode_dynamic_bert_tutorial.rst:182]: Nit: Add "the" - Right now qconfig_dict is "the" only way...

Comment (jlin27) [reply prototype_source/graph_mode_dynamic_bert_tutorial.rst:182]: Nit: Capitalize "for" - "For" example, if we...

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+
+.. code::
+
+    Running benchmark for Float model
+    Elapsed time: 4.49 s
+    Running benchmark for Eager Mode Quantized model
+    Elapsed time: 2.67 s
+    Running benchmark for Graph Mode Quantized model
+    Elapsed time: 2.69 s
+    As we can see both graph mode and eager mode quantized model have a similar speed up over the floating point model.
```
Comment (jlin27) [inline prototype_source/graph_mode_dynamic_bert_tutorial.rst:551]: Suggestion: Add a conclusion section with links to references or learn more at the end. Example:
https://pytorch.org/tutorials/intermediate/dynamic_quantization_bert_tutorial.html#conclusion

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,551 @@
+(prototype) Graph Mode Dynamic Quantization on BERT
+==============================================
+
+
+**Author**: `Supriya Rao <https://github.com/supriyar>`_
+
+Introduction
+------------
+
+This tutorial introduces the steps to do post training Dynamic Quantization with Graph Mode Quantization. Dynamic quantization converts a float model to a quantized model with static int8 data types for the weights and dynamic quantization for the activations. The activations are quantized dynamically (per batch) to int8 while the weights are statically quantized to int8. Graph Mode Quantization flow operates on the model graph and requires minimal user intervention to quantize the model. To be able to use graph mode, the float model needs to be either traced or scripted first.
```
Comment (jlin27) [inline prototype_source/graph_mode_dynamic_bert_tutorial.rst:10]: Is there documentation/doc stings or source code we can link to throughout this tutorial or at least in the introduction for users to know where to find additional info?

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst (hunk 1 above)
Comment (supriyar) [reply prototype_source/graph_mode_dynamic_bert_tutorial.rst:18]: The tutorial will be released along with this one. How do I link it in that case?

File: prototype_source/graph_mode_dynamic_bert_tutorial.rst (hunk 4 above)
Comment (supriyar) [reply prototype_source/graph_mode_dynamic_bert_tutorial.rst:10]: Sure, how do I link to something that hasn't been released yet? For example `:func:`~torch.quantization.quantize_dynamic_jit` in https://github.com/pytorch/pytorch/blob/master/docs/source/quantization.rst#id153

Comment (jlin27) [reply prototype_source/graph_mode_dynamic_bert_tutorial.rst:10]: Could you link to the sub-header link for now?
https://pytorch.org/docs/master/quantization.html#torch-quantization

or could you link to the source code? 
https://github.com/pytorch/pytorch/blob/master/torch/quantization/quantize_jit.py

**Pull request merged**